### PR TITLE
Nodejs/week2

### PR DESCRIPTION
**Pull request by GreeshmaRahool**
Added nodejs/week2 homework

### CODE REVIEW

File: nodejs/week2/warmup/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const { response } = require("express");
+const express = require("express");
+const app = express();
+const numbersRouter = require('./numbersRouter');
+
+app.get("/", (request, response) => response.send("nodejs week2 homework"));
+
+app.use('/numbers', numbersRouter);
```
Comment (Laila1222) [inline nodejs/week2/warmup/index.js:8]: You are missing app.listen from this file. Without that, the server is not going to 'listen' for the requests. 
app.listen(3000, () => console.log(`Calculator:listening on port 3000`));

File: nodejs/week2/warmup/numbersRouter.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    if (params.first && params.second) {
+
+        const product = parseInt(params.first) * parseInt(params.second);
+        response.send(`The product is ${product}`);
+       
+    }
+    else {
+        
+        response.status(400).send('Bad request');
+    }
```
Comment (Laila1222) [inline nodejs/week2/warmup/numbersRouter.js:39]: Good!

File: nodejs/week2/meal-sharing/src/backend/api/meals-router.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        return response.status(400).send("Invalid limit");
+      }
+      response.send(meals.slice(0, inputLimit));
+    } else {
+      response.send(meals);
+    }
+  } catch (error) {
+    response.status(400).send("Bad request!");
+  }
+});
```
Comment (Laila1222) [inline nodejs/week2/meal-sharing/src/backend/api/meals-router.js:49]: Well done

File: nodejs/week2/meal-sharing/src/backend/api/reviews-router.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    if (getReviewWithId.length === 0) {
+      return response.status(404).send("No data found!");
+    }
+    if (isNaN(parseInt(params.id))) {
+      return response.status(400).send("Not a valid id");
+    }
+  } catch (error) {
+    response.status(400).send("Bad request!");
+  }
+});
```
Comment (Laila1222) [inline nodejs/week2/meal-sharing/src/backend/api/reviews-router.js:33]: Good

File: nodejs/week2/meal-sharing/src/backend/api/reservations-router.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    if (getReservationWithId.length === 0) {
+      return response.status(404).send("No data found!");
+    }
+    if (isNaN(parseInt(params.id))) {
+      return response.status(400).send("Not a valid id");
+    }
+  } catch (error) {
+    response.status(400).send("Bad request!");
+  }
+});
```
Comment (Laila1222) [inline nodejs/week2/meal-sharing/src/backend/api/reservations-router.js:32]: Well done

**Review by Laila1222 — APPROVED** [review]
Well done! Very good job. In the warmup you are missing an important functionality (see comment), but otherwise great!

File: nodejs/week2/warmup/index.js (hunk 1 above)
Comment (GreeshmaRahool) [reply nodejs/week2/warmup/index.js:8]: Thank you for the great comments. I have added the app. listen part :)

**Pull request merged**